### PR TITLE
docs: Add how-to guide for `SQLDatabaseLoader`

### PR DESCRIPTION
**Pull request by amotl**
- **Description:** Documentation about the generic document loader adapter for SQLAlchemy, `SQLDatabaseLoader`.
  - **Needed by:** https://github.com/crate-workbench/langchain/pull/1
  - **Followup from:** https://github.com/langchain-ai/langchain/pull/18281
  - **Addressed to:** @eyurtsev

Dear Eugene,

coming from GH-16246/GH-18281, this patch adds missing documentation that has still been included in https://github.com/crate-workbench/langchain/pull/1. We wanted to break it out of that patch, in order to further minimise its size before submitting it.

With kind regards,
Andreas.

NB: Can I also humbly ask you to review/exercise this for validity, @simonprickett or @kneth? I think the demonstrated code should work with recent vanilla versions of LangChain, with all of SQLite, PostgreSQL, and CrateDB.

### CODE REVIEW

File: docs/docs/how_to/document_loader_sql_database.ipynb [hunk 1]
```diff
@@ -0,0 +1,360 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SQL Database\n",
```
Comment (amotl) [inline docs/docs/how_to/document_loader_sql_database.ipynb:7]:
```suggestion
    "# SQLDatabaseLoader\n",
```

File: docs/docs/how_to/document_loader_sql_database.ipynb [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    "\n",
+    "## About\n",
+    "\n",
+    "The `SQLDatabaseLoader` loads records from any database supported by\n",
+    "[SQLAlchemy], see [SQLAlchemy dialects] for the whole list of supported\n",
+    "SQL databases and dialects.\n",
+    "\n",
+    "For talking to the database, the document loader uses the [SQLDatabase]\n",
+    "utility from the LangChain integration toolkit.\n",
```
Comment (amotl) [inline docs/docs/how_to/document_loader_sql_database.ipynb:16]:
```suggestion
    "For talking to the database, the document loader uses the [SQLDatabase]\n",
    "utility.\n",
```

File: docs/docs/how_to/document_loader_sql_database.ipynb [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    "## What's inside\n",
+    "\n",
+    "This notebook covers how to load documents from an [SQLite] database,\n",
+    "using the [SQLAlchemy] document loader.\n",
+    "\n",
+    "It loads the result of a database query with one document per row.\n",
+    "\n",
+    "[SQLAlchemy]: https://www.sqlalchemy.org/\n",
+    "[SQLAlchemy dialects]: https://docs.sqlalchemy.org/en/latest/dialects/\n",
+    "[SQLDatabase]: https://python.langchain.com/docs/integrations/toolkits/sql_database\n",
```
Comment (amotl) [inline docs/docs/how_to/document_loader_sql_database.ipynb:35]: Linking to the [SQLDatabase Toolkit](https://python.langchain.com/docs/integrations/tools/sql_database/) was probably wrong?

```suggestion
    "[SQLDatabase]: https://python.langchain.com/api_reference/community/utilities/langchain_community.utilities.sql_database.SQLDatabase.html\n",
```

**Review by amotl — COMMENTED** [review]
Submitting just a few comments from self-review.

**Comment by eyurtsev** [comment]
Hi @amotl, thank you for taking the initiative to improve the how-to guides!

We don't have good guidelines yet for the how-to guides, but if you can browse through a few (https://python.langchain.com/docs/how_to/), you may get a bit of a sense of the format:

1. Use a notebook format only here (remove the mdx file).
2. Add a link to the notebook from the how-to guide index in an appropriate place
3. The title should be phrased as a how-to question (e.g., "How to ...")
4. If you can link to conceptual prerequisites (e.g., https://python.langchain.com/docs/how_to/passthrough/). Potentially OK to link to any SQL resources that make sense for folks to know about
5. Don't comment out the package install (maybe simply skip it if you already have the packages installed)
6. Review https://diataxis.fr/how-to-guides/ if you're not familiar with diataxis

**Review by kneth — APPROVED** [review]
LGTM

I would appreciate additional reviewers

**Comment by eyurtsev** [comment]
Hi @efriis re-assigned to you. I'm not sure if this PR is being actively worked on right now (probably on the back burner). But I check the integrations page https://python.langchain.com/docs/integrations/document_loaders/ and don't think we have integration docs for this loader.

I suspect we'll want to match this pattern: https://python.langchain.com/docs/integrations/document_loaders/

**Comment by efriis** [comment]
ah yep I'll close this one and discuss with their team on #27710 

I think they put up all the PRs at once hence the confusing sequencing, and the new integration process will be much better suited to this!

this one can be reopened after the integration is done/published/docs are in

**Comment by amotl** [comment]
Hi. I think this one can stay, as it improves existing functionality with corresponding documentation, that has been missed on the first iteration.

**Comment by efriis** [comment]
ah I misunderstood. @amotl can we
- remove the changes to the docs/docs/how_to directory
- instead add a single ipynb page to docs/integrations/document_loaders/sql_database or something like that (matching the format eugene linked)

**Comment by amotl** [comment]
Hi Erick. Sure. As soon as I will resume working on this patch, I will commandeer it into the directions you are suggesting. Thank you very much.

**Comment by efriis** [comment]
great! if it's not in the next week, would recommend closing and reopening when you're ready for a review!

**Comment by amotl** [comment]
Hi Erick. Sure. Let's close, and re-open later.

**Pull request closed without merging**